### PR TITLE
Add testing for zero tensors in PyTorch backend

### PR DESCRIPTION
**Pull request by Tabrizian**
Testing for https://github.com/triton-inference-server/pytorch_backend/pull/120

### CODE REVIEW

**Review by oandreeva-nv — DISMISSED** [review]
Nice refactoring!

File: qa/L0_infer_zero/infer_zero_test.py [hunk 1]
```diff
@@ -62,72 +79,19 @@ def _full_zero(self, dtype, shapes):
                 ]
                 iu.infer_zero(
                     self,
-                    "graphdef",
+                    backend,
                     bs,
```
Comment (rmccorm4) [inline qa/L0_infer_zero/infer_zero_test.py:83]: will `_nobatch` models fail for bs 8?

Comment (rmccorm4) [reply qa/L0_infer_zero/infer_zero_test.py:83]: oh nvm, I see the distinction with `${backend}_nobatch` below.

File: qa/L0_infer_zero/infer_zero_test.py [hunk 2]
```diff
@@ -41,17 +41,34 @@
...
         # 'shapes' is list of shapes, one for each input.
+        for backend in BACKENDS.split(" "):
+            if backend == "graphdef" or backend == "savedmodel":
+                if not tu.validate_for_tf_model(
+                    dtype, dtype, dtype, shapes[0], shapes[0], shapes[0]
+                ):
+                    return
+
+            if backend == "onnx":
+                if not tu.validate_for_tf_model(
```
Comment (rmccorm4) [inline qa/L0_infer_zero/infer_zero_test.py:58]: validate_for_onnx_model?

Comment (Tabrizian) [reply qa/L0_infer_zero/infer_zero_test.py]: good catch, fixed.

Comment (rmccorm4) [reply qa/L0_infer_zero/infer_zero_test.py]: A further refactor could be something like this, if you think it would be a nice change:
```python
VALIDATION_FNS = {
  "onnx": tu.validate_for_onnx_model,
  # ...
}

# ...

            # object models do not exist right now for PyTorch
            if backend == "libtorch" and dtype == "object":
                    return

            if not VALIDATION_FNS[backend](dtype, dtype, dtype, shapes[0], shapes[0], shapes[0]):
                return
```

Up to you though

**Review by kthui — DISMISSED** [review]

File: qa/L0_infer_zero/infer_zero_test.py (hunk 2 above)
Comment (Tabrizian) [reply qa/L0_infer_zero/infer_zero_test.py]: good suggestion, done.

**Review by kthui — APPROVED** [review]

**Review by rmccorm4 — APPROVED** [review]
Nice cleanup 🎉
